app/assets/src/components/PipelineSampleReport.jsx

```diff
@@ -702,87 +702,39 @@ class PipelineSampleReport extends React.Component {
   };
 
   displayHoverActions = (taxInfo, reportDetails) => {
-    let tax_level_str = "";
-    let ncbiDot, fastaDot, alignmentVizDot, phyloTreeDot, contigVizDot;
-    if (taxInfo.tax_level == 1) tax_level_str = "species";
-    else tax_level_str = "genus";
-
-    let valid_tax_id =
+    const validTaxId =
       taxInfo.tax_id < this.INVALID_CALL_BASE_TAXID || taxInfo.tax_id > 0;
-    if (valid_tax_id)
-      ncbiDot = (
-        <i
-          data-tax-id={taxInfo.tax_id}
-          onClick={this.gotoNCBI}
-          className="fa fa-link action-dot"
-          aria-hidden="true"
-        />
-      );
-    if (reportDetails.taxon_fasta_flag)
-      fastaDot = (
-        <i
-          data-tax-level={taxInfo.tax_level}
-          data-tax-id={taxInfo.tax_id}
-          onClick={this.downloadFastaUrl}
-          className="fa fa-download action-dot"
-          aria-hidden="true"
-        />
-      );
-    if (this.canSeeAlignViz && valid_tax_id && taxInfo.NT.r > 0)
-      alignmentVizDot = (
-        <i
-          data-tax-level={tax_level_str}
-          data-tax-id={taxInfo.tax_id}
-          onClick={this.gotoAlignmentVizLink}
-          className="fa fa-bars action-dot"
-          aria-hidden="true"
-        />
-      );
-    if (this.state.contigTaxidList.indexOf(taxInfo.tax_id) >= 0)
-      contigVizDot = (
-        <i
-          data-tax-id={taxInfo.tax_id}
-          onClick={this.downloadContigUrl}
-          className="fa fa-puzzle-piece action-dot"
-          aria-hidden="true"
-        />
-      );
-    if (
+    const ncbiEnabled = validTaxId;
+    const fastaEnabled = reportDetails.taxon_fasta_flag;
+    const contigVizEnabled =
+      this.state.contigTaxidList.indexOf(taxInfo.tax_id) >= 0;
+    const alignmentVizEnabled =
+      this.canSeeAlignViz && validTaxId && taxInfo.NT.r > 0;
+    const phyloTreeEnabled =
       this.allowPhyloTree &&
       taxInfo.tax_id > 0 &&
-      PhyloTreeChecks.passesCreateCondition(taxInfo.NT.r, taxInfo.NR.r)
-    )
-      phyloTreeDot = (
-        <PhyloTreeCreationModal
```

Comment:
Was this creating the Modal for every row?

Reply:
Hmmm I guess so!